pkg/solana/txm/txm.go

```diff
@@ -135,14 +140,18 @@ func (txm *Txm) run() {
 				continue           // skip remainining
 			}
 
-			// send tx + signature to simulation queue
-			msg.tx = &tx
-			msg.signature = sig
-			msg.id = id
-			select {
-			case txm.chSim <- msg:
-			default:
-				txm.lggr.Warnw("failed to enqeue tx for simulation", "queueFull", len(txm.chSend) == MaxQueueLen, "tx", msg)
+			// Only add tx to simulation queue if the EstimateComputeUnitLimit feature is disabled
+			// Otherwise, the tx would have been simulated before broadcast
+			if !msg.cfg.EstimateComputeUnitLimit {
+				// send tx + signature to simulation queue
+				msg.tx = &tx
+				msg.signature = sig
+				msg.id = id
+				select {
+				case txm.chSim <- msg:
+				default:
+					txm.lggr.Warnw("failed to enqeue tx for simulation", "queueFull", len(txm.chSend) == MaxQueueLen, "tx", msg)
+				}
```

Comment:
Could a second simulation pick up an error that the original didn't? I was thinking if a transaction passes simulation for the compute unit limit then timeout or confirmation should be the only reasons to cancel retry. I guess it doesn't hurt to keep it in case there's some state change in between

Reply:
hmmmm - fair point

technically we are modifying the transaction with the limit and the price - so technically it could still fail if estimation/tx building for either was broken 🤔 - unlikely but possible

Reply:
Oh ya good point. Makes sense to keep it then. I've undone this in the latest commit